src/workloads/execution/SSBColumnStoreIndex.yml

```diff
@@ -0,0 +1,595 @@
+            "c_nation": "$customer.nation.n_name",
+            "p_brand": "$part.p_brand"
+          },
+          "revenue": {
+            "$sum": {
+              "$subtract": [
+                "$l_extendedprice",
+                {"$multiply": ["$l_extendedprice", "$l_discount"]}
+              ]
+            }
+          }
+        }
+      },
+      {
+        "$addFields": {
+          "order_year": "$_id.order_year",
+          "customer_nation": "$_id.c_nation",
+          "part_brand": "$_id.p_brand"
+        }
+      },
+      {"$sort": {"order_year": 1, "customer_nation": 1, "part_brand": 1}}
+    ]
+  cursor: {}
+
+# Defines test operations.
+# Here we use explain with verbosity = execustionStats instead of using plain aggregate command
+# so that aggregate pipeline runs to completion but yet we don't need to exhaust cursor. As of now
+# we don't have an easy way to exhaust cursor.
+TestOperations: &TestOperations
+- OperationMetricsName: Q1_1Query
+  OperationName: RunCommand
+  OperationCommand:
+    explain: *Q1_1Query
+    verbosity:
+      executionStats
+- OperationMetricsName: Q1_2Query
+  OperationName: RunCommand
+  OperationCommand:
+    explain: *Q1_2Query
+    verbosity:
+      executionStats
+- OperationMetricsName: Q1_3Query
+  OperationName: RunCommand
+  OperationCommand:
+    explain: *Q1_3Query
+    verbosity:
+      executionStats
+- OperationMetricsName: Q2_1Query
+  OperationName: RunCommand
+  OperationCommand:
+    explain: *Q2_1Query
+    verbosity:
+      executionStats
+- OperationMetricsName: Q2_2Query
+  OperationName: RunCommand
+  OperationCommand:
+    explain: *Q2_2Query
+    verbosity:
+      executionStats
+- OperationMetricsName: Q2_3Query
+  OperationName: RunCommand
+  OperationCommand:
+    explain: *Q2_3Query
+    verbosity:
+      executionStats
+- OperationMetricsName: Q3Query
+  OperationName: RunCommand
+  OperationCommand:
+    explain: *Q3Query
+    verbosity:
+      executionStats
+- OperationMetricsName: Q4_1Query
+  OperationName: RunCommand
+  OperationCommand:
+    explain: *Q4_1Query
+    verbosity:
+      executionStats
+- OperationMetricsName: Q4_2Query
+  OperationName: RunCommand
+  OperationCommand:
+    explain: *Q4_2Query
+    verbosity:
+      executionStats
+- OperationMetricsName: Q4_3Query
+  OperationName: RunCommand
+  OperationCommand:
+    explain: *Q4_3Query
+    verbosity:
+      executionStats
+
+# Defines how many times each test operation is executed.
+TestRepeatCount: &TestRepeatCount 1
+
+SetupPhase: &SetupPhase 0
+BuildColumnStoreIndexPhase: &BuildColumnStoreIndexPhase 1
+ColumnstoreScanPhase: &ColumnstoreScanPhase 2
+DropColumnstoreIndexPhase: &DropColumnstoreIndexPhase 3
+CollScanPhase: &CollScanPhase 4
+MaxPhases: &MaxPhases 5
+
+ColumnStoreIndexName: &ColumnStoreIndexName $**_columnstore
+
+Actors:
+- Name: Setup
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [*SetupPhase]
+      NopInPhasesUpTo: *MaxPhases
```

Comment:
For my education: is the `NopInPhaseUpTo` field still required when `Active` is provided?

Reply:
Genny crashes if you omit it, so I'd say yes.

Reply:
Hmm,  I think that it should be possible for genny to calculate MaxPhases automatically if is is not supplied or at least fail with a reasonable error message (rather than crashing). I've created [PERF-3536](https://jira.mongodb.org/browse/PERF-3536) to track this.